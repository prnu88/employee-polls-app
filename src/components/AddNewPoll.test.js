import AddNewPoll from './AddNewPoll';
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import { fireEvent, render } from "@testing-library/react";
import thunk from 'redux-thunk';
import { MemoryRouter } from 'react-router';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const store = mockStore({authedUser:'mtsamis',users: {mtsamis: {
  id: 'mtsamis',
  password:'xyz123',
  name: 'Mike Tsamis',
  avatarURL: "https://image.freepik.com/free-psd/3d-female-character-holding-tablet-device_23-2148938895.jpg",
  answers: {
    "xj352vofupe1dqz9emx13r": 'optionOne',
    "vthrdm985a262al8qx3do": 'optionTwo',
    "6ni6ok3ym7mf1p33lnez": 'optionOne'
  },
  questions: ['6ni6ok3ym7mf1p33lnez', 'xj352vofupe1dqz9emx13r'],
}}});

describe("AddNewPoll Component", () => {
    it("AddNewPoll should render without throwing an error", () => {
        expect(
          <Provider store={store}>
          <AddNewPoll />
          </Provider>
        ).toMatchSnapshot();
    });

    it("Adding New Poll clears the form", async() => {
      var component = render(
        <Provider store={store}><MemoryRouter><AddNewPoll /></MemoryRouter></Provider>
      );
      
      var inputFirstOption = component.getByTestId('input-first-option');
      var inputSecondOption = component.getByTestId('input-second-option');

      fireEvent.change(inputFirstOption, {target: {value: 'buy a cat'}})
      fireEvent.change(inputSecondOption, {target: {value: 'buy a dog'}})

      var submitBtn = component.getByTestId('add-poll-btn');
      fireEvent.click(submitBtn);

      expect(inputFirstOption.value).toEqual("");
      expect(inputSecondOption.value).toEqual("");
  });
});


