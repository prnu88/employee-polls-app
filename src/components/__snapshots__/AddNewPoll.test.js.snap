// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNewPoll Component AddNewPoll should render without throwing an error 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Memo(Connect(AddNewPoll)) />
</Provider>
`;
